### PR TITLE
Task-50800: Implement edit target service

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
Prior to this change, only frontend part of edit target is implemented. After this change, backend implementation is done which makes the edit target feature available.

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/service/NewsTargetingService.java [hunk 1]
```diff
@@ -103,14 +103,29 @@
...
+   * @param currentIdentity current {@link Identity} attempting to create {@link News} target
+   * 
+   * @return created {@link News} target {@link Metadata}
+   * @throws IllegalArgumentException when user creates a {@link News} target that already exists
    * @throws IllegalAccessException when user doesn't have access to create {@link News} target
    */
   Metadata createNewsTarget(NewsTargetingEntity newsTargetingEntity, org.exoplatform.services.security.Identity currentIdentity) throws IllegalArgumentException, IllegalAccessException;
 
+  /**
+   * Update news Metadata by it name
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/NewsTargetingService.java]:
```suggestion
   * Update news target
```

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 2]
```diff
@@ -214,4 +214,4 @@ news.publishTargets.managementDrawer.btn.confirm=Confirm
 news.publishTargets.managementDrawer.btn.cancel=Cancel
 news.publishTargets.managementDrawer.placeholder.name=Enter the target name
 news.publishTargets.managementDrawer.placeholder.description=Enter a description here
-news.publishTargets.managementDrawer.sameNewsTargetWarning=A target with the same name already exists.
+news.publishTargets.managementDrawer.sameNewsTargetWarning=A target with the same name already exists.
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]: not changes, revert this

File: webapp/src/main/webapp/news-list-view/components/settings/NewsSettingsDrawer.vue [hunk 3]
```diff
@@ -138,11 +138,7 @@ export default {
       return this.viewTemplates.filter(e=> !e.name.includes('EmptyTemplate'));
     },
     checkAlphanumeric() {
-      if (this.newsHeader && !this.newsHeader.trim().match(/^[\w\-\s]+$/) && this.newsHeader.length > 0) {
-        return this.$t('news.list.settings.name.errorMessage');
-      } else {
-        return '';
-      }
+      return this.targetName && !this.targetName.trim().match(/^[\w\-\s]+$/) && this.targetName.length > 0 ? this.$t('news.list.settings.name.errorMessage') : '';
```
Comment (azayati) [inline webapp/src/main/webapp/news-list-view/components/settings/NewsSettingsDrawer.vue:141]: this modification is a part of develop one https://github.com/exoplatform/news/commit/2c66a5445554b672c7426a42eda0ff810d026242#diff-7853b826a9d2c891b72fcf7bb148c5fde29d6335e95dc4c3a64b1cae755ccab2R104, so to be reverted

File: webapp/src/main/webapp/services/newsTargetingService.js [hunk 4]
```diff
@@ -83,3 +83,20 @@ export function createTarget(target) {
...
+    credentials: 'include',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(target),
+  }).then((resp) => {
+    if (resp && resp.ok) {
+      return resp.status;
+    } else {
+      throw new Error('Error updating news target');
```
Comment (azayati) [inline webapp/src/main/webapp/services/newsTargetingService.js]:
```suggestion
      throw new Error('Error when updating news target');
```

File: services/src/main/java/org/exoplatform/news/rest/NewsTargetingRestResourcesV1.java [hunk 5]
```diff
@@ -207,34 +204,62 @@ public Response undoDeleteTarget(@Context HttpServletRequest request,
...
+  @ApiResponses( value = {
+    @ApiResponse(code = HTTPStatus.NOT_FOUND, message = "Object not found"),
+    @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+    @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"),
+    @ApiResponse(code = HTTPStatus.CONFLICT, message = "Conflict operation")
+  })
+  public Response updateNewsTarget(@ApiParam(value = "News target to create", required = true)
+                                   NewsTargetingEntity newsTargetingEntity,
+                                   @ApiParam(value = "Original news target name", required = true)
+                                   @QueryParam("originalTargetName") String originalTargetName) {
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/rest/NewsTargetingRestResourcesV1.java]: PathParam is better IMO

**Review by azayati — CHANGES_REQUESTED** [review]
see review

File: webapp/src/main/webapp/news-list-view/components/settings/NewsSettingsDrawer.vue (hunk 3 above)
Comment (hassenelkamel-exo) [reply webapp/src/main/webapp/news-list-view/components/settings/NewsSettingsDrawer.vue:141]: not the same file, in order to optimize code we add it

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

**Review by azayati — DISMISSED** [review]
LGTM

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**